data_steward/analytics/cdr_ops/ehr_retraction_validation.py

```diff
@@ -0,0 +1,216 @@
+# -
+
+# ## 2. Verify Row counts of participants in source dataset minus the ehr data is equal to the row counts for the participants in new dataset.
+
+# +
+table_row_counts_query = JINJA_ENV.from_string('''
+SELECT
+\'{{table_name}}\' as table_name,
+old_minus_ehr_row_count,
+new_row_count,
+case when old_minus_ehr_row_count = new_row_count then 'OK'
+  when old_minus_ehr_row_count is null AND new_row_count is null then 'OK'
+  when old_minus_ehr_row_count is NOT NULL and new_row_count is null then 'PROBLEM'
+  when old_minus_ehr_row_count is NULL and new_row_count is not NULL then 'PROBLEM'
+  ELSE 'PROBLEM'
+  end as table_count_status FROM
+(SELECT
+  count(*) as old_minus_ehr_row_count,
+  (select count(*)
+  from `{{project}}.{{new_dataset}}.{{table_name}}` tb1
+  inner join pids using (person_id) 
+  ) as new_row_count
+FROM
+  `{{project}}.{{old_dataset}}.{{table_name}}` as tb
+INNER JOIN
+  pids as p USING (person_id)
+INNER join `{{project}}.{{old_dataset}}.{{mapping_table}}` mp
+on tb.{{table_name}}_id = mp.{{table_name}}_id
+{% if is_deidentified == 'true' %}
+where src_id = 'PPI/PM'
+{% else %}
+where src_hpo_id = 'rdr'
+{% endif %}
+)
+''')
+row_counts_queries_list = []
+for table in pid_table_list:
+    row_counts_queries_list.append(
+        table_row_counts_query.render(project=project_id,
+                                      old_dataset=old_dataset,
+                                      new_dataset=new_dataset,
+                                      is_deidentified=is_deidentified,
+                                      table_name=table,
+                                      mapping_table=mapping_table_for(
+                                          table, is_deidentified)))
+
+row_counts_union_all_query = '\nUNION ALL\n'.join(row_counts_queries_list)
+
+retraction_table_count_query = (f'{rids_query}\n{row_counts_union_all_query}'
+                                if is_deidentified == 'true' else
+                                f'{pids_query}\n{row_counts_union_all_query}')
+execute(client, retraction_table_count_query)
+# -
+
+# ## 3. Verify Death table retraction.
+
+# +
+table_check_query = JINJA_ENV.from_string('''
+SELECT
+  'Death' AS table_name,
+  Case when count(tb.person_id) = 0 then 'OK'
+  ELSE 
+  'PROBLEM' end as retraction_status
+FROM
+  `{{project}}.{{dataset}}.death` as tb
+right JOIN
+  pids as p
+USING(person_id)
+''')
+death_query = table_check_query.render(project=project_id,
+                                       dataset=new_dataset,
+                                       table_name=table)
+
+retraction_status_query = (f'{rids_query}\n{death_query}'
+                           if is_deidentified == 'true' else
+                           f'{pids_query}\n{death_query}')
+execute(client, retraction_status_query)
+# -
+
+# ## 4. Verify if mapping/ext tables are cleaned after retraction
+
+# +
+mapping_ext_check_query = JINJA_ENV.from_string('''
+select \'{{mapping_table}}\' as table_name,
```

Comment:
Can this check be made to resemble the other checks with a status column?  You could just add a case statement to this query.